### PR TITLE
[Python PQ/RQ] Functional

### PR DESCRIPTION
**Pull request by sebaraba**
## Done:

- [x] benefits-of-using-generators
- [x] functools-wraps
- [x] lambda-functions

### CODE REVIEW

**Comment by enki-hq** [comment]
### Insight previews:
- [ ] https://enkipro.compreview-insight/enkidevs/curriculum/python/functional-programming/decorators/functools-wraps.md?ref=21d4883d065c6a4d141ad5ae3382a9f9a52387d0
- [ ] https://enkipro.compreview-insight/enkidevs/curriculum/python/functional-programming/functional-programming/lambda-functions.md?ref=21d4883d065c6a4d141ad5ae3382a9f9a52387d0
- [ ] https://enkipro.compreview-insight/enkidevs/curriculum/python/functional-programming/generators/benefits-of-using-generators.md?ref=21d4883d065c6a4d141ad5ae3382a9f9a52387d0

File: python/functional-programming/decorators/functools-wraps.md [hunk 1]
```diff
@@ -57,6 +57,17 @@ print(say_hello.__doc__)
 # docstring
 ```
 
+---
+## Practice
+
+What is an improtant drawback of `decorators`?
```
Comment (lizTheDeveloper) [inline python/functional-programming/decorators/functools-wraps.md]: eh... let's ask them to use a generator here

File: python/functional-programming/functional-programming/lambda-functions.md [hunk 2]
```diff
@@ -50,6 +50,21 @@ In this case, we are passing the lambda function `x % 3 == 0` to `filter()` to b
...
+
+```python
+
+l = [-1,-2,3,-4,5,6]
+
+new = filter(??? x: ???, ???)
+```
+* lambda
+* x > 0
+* l
```
Comment (lizTheDeveloper) [inline python/functional-programming/functional-programming/lambda-functions.md]: add some wrong answers here

File: python/functional-programming/generators/benefits-of-using-generators.md [hunk 3]
```diff
@@ -40,6 +40,14 @@ Generators are functions which behave like iterators, but can have better perfor
 
 However, the set of generated values can only be used once.
 
+---
+## Practice
+
+When using generators ??? wait for all the values to be generated before using them.
```
Comment (lizTheDeveloper) [inline python/functional-programming/generators/benefits-of-using-generators.md]: I think this one needs a code sample, and a pythontutor link to really explain it

File: python/functional-programming/functional-programming/lambda-functions.md (hunk 2 above)
Comment (sebaraba) [reply python/functional-programming/functional-programming/lambda-functions.md]: I added more answers here, it is just github that doesn't compress this comment.

**Pull request merged**